src/thunderbots/ai/world/field.h

```diff
@@ -0,0 +1,243 @@
+#ifndef AI_FIELD_H_
+#define AI_FIELD_H_
+
+#include "geom/point.h"
+#include "thunderbots_msgs/Field.h"
+
+/**
+ * Exposes the dimensions of various parts of the field.
+ */
+class Field
+{
+   public:
+    /**
+     * Constructs a new field
+     */
+    explicit Field();
+
+    /**
+     * Updates the dimensions of the field given a field message. All dimensions should be
+     * in metres.
+     *
+     * @param new_field_msg the field message containing the new field information
+     */
+    void updateDimensions(thunderbots_msgs::Field new_field_msg);
+
+    /**
+     * Updates the dimensions of the field. All values should be in metres.
+     *
+     * @param field_length the length of the playing area (along the x-axis)
```

Comment:
nit: would be convenient to include units in here as well

Reply:
units are mentioned in the first line (All units should be in metres). I didn't want to type it out for every line.